_client/src/Dialogs/DialogManager.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QString>
+#include <QObject>
+#include <QMessageBox>
+
+#include "LoginDialog.hpp"
+#include "RegistrationDialog.hpp"
+#include "Network/ApiManager.hpp"
+
+/**
+ * @class DialogManager
+ * Controls all dialogs in MessageBoxes
+ */
+class DialogManager : public QObject
+{
+    Q_OBJECT
+
+protected:
+public:
+    /**
+     * @brief Constructor and destructor of DialogManager, which control other objects
+     */
+    DialogManager(ApiManager *apiManager);
+    virtual ~DialogManager();
+
+    /**
+     * @brief Show MessageBox with error message
+     * @param message is text of error
+     */
+    void showErrorMessageBox(const QString &message);
+
+    /**
+     * @brief Show LoginDialog object
+     */
+    void showLoginDialog();
+
+    /**
+     * @brief Show RegistrationDialog object
+     */
+    void showRegistrationDialog();
+
+private:
+    /**
+     * @brief initialization of dialogs
+     */
+    void initDialogs();
+
+    /**
+     * @brief setup connections for ApiManager
```

Comment:
Not for ApiManager, but between ApiManager and dialogs

Reply:
changed